src/vs/workbench/electron-browser/workbench.ts

```diff
@@ -925,6 +949,15 @@ export class Workbench extends Disposable implements IPartService {
 			this.notificationsCenter,
 			this.notificationsToasts
 		);
+
+		if (!isMacintosh && this.getCustomTitleBarStyle()) {
+			this.windowService.isMaximized().then((max) => {
```

Comment:
@sbatten I am not a big fan of the various calls to `windowService.isMaximized()` on startup because this runs async while the workbench is being created and can potentially lead into flicker. I would suggest to carry over a property into the `IWindowConfiguration` that has this information already in so that you do not have to call this on startup async. 

The layout should then just receive this as part of creation and do the right thing.

Reply:
I understand the complaint, but I don't quite understand the solution. The window service that holds the configuration still has to make an async call into the windowsService it holds

Reply:
@sbatten I suggest to transport this state over all the way from the place where the window loads the URL in the browser process: https://github.com/Microsoft/vscode/blob/master/src/vs/code/electron-main/window.ts#L521

It should be possible to know the initial state there and then you do not need to make an async call on workbench creation to query it. The config object is accessible for anyone via [IWindowService#getConfiguration()](https://github.com/Microsoft/vscode/blob/master/src/vs/platform/windows/electron-browser/windowService.ts#L36)